File: EvenementsEnDirect/src/www/classes/Event.php
<?php
/* Owner : Davila Lou IDAP4A
*  Project : Live Events (TPI 2020)
*  Version : 1.0
*  Date : 25/05/2020 - 09/06/2020
*/

/**
 * This class contains the event informations we get from database
 */
class Event
{
    /**
     * Id of the event
     *
     * @var int
     */
    public $id;
    /**
     * Title of the event
     *
     * @var string
     */
    public $title;
    /**
     * Description of the event
     *
     * @var string
     */
    public $description;
    /**
     * The starting date and time of the event
     *
     * @var string
     */
    public $startDateTime;
    /**
     * The ending date and time of the event
     *
     * @var string
     */
    public $endDateTime;
    /**
     * The current state of the event
     *
     * @var string
     */
    public $state;
    /**
     * Is the event visible
     *
     * @var boolean
     */
    public $isVisible;
    /**
     * Le pays de l'événement
     *
     * @var int
     */
    public $country;
    /**
     * Construct
     *
     * @param int $id
     * @param string $title
     * @param string $description
     * @param string $state
     * @param string $country
     * @param string $startDateTime
     * @param string $endDateTime
     * @param int $isVisible
     */
    public function __construct($id,$title,$description,$state,$country,$startDateTime,$endDateTime = null,$isVisible='1')
    {
        $this->id = $id;
        $this->title = $title;
        $this->description = $description;
        $this->startDateTime = $startDateTime;
        $this->endDateTime = $endDateTime;
        $this->state = $state;
        $this->isVisible = $isVisible;
        $this->country = $country;
    }
}
?>

File: EvenementsEnDirect/src/mobileApp/LiveEvents/Event.cs
/* Owner : Davila Lou IDAP4A
*  Project : Live Events (TPI 2020)
*  Version : 1.0
*  Date : 25/05/2020 - 09/06/2020
*/
using System;
using System.Collections.Generic;
using System.Runtime.Remoting.Messaging;

namespace LiveEvents
{
    /// <summary>
    /// This class contains the informations of an event
    /// </summary>
    public class Event
    {
        /// <summary>
        /// The id of the event
        /// </summary>
        public int Id { get; }
        /// <summary>
        /// The title of the event
        /// </summary>
        public string Title { get; }
        /// <summary>
        /// The description of the event
        /// </summary>
        public string Description { get; }
        /// <summary>
        /// The state of the event
        /// </summary>
        public string State { get; }
        /// <summary>
        /// The country of the event
        /// </summary>
        public string Country { get; }
        /// <summary>
        /// The start date and time of the event
        /// </summary>
        public DateTime StartDateTime { get; }
        /// <summary>
        /// The end date and time of the event
        /// </summary>
        public DateTime EndDateTime { get; }
        /// <summary>
        /// The messages of the event
        /// </summary>
        public List<Message> Messages { get; set; }

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="id">The id of the event</param>
        /// <param name="title">The title of the event</param>
        /// <param name="description">The description of the event</param>
        /// <param name="startDateTime">The starting datetime of the event</param>
        /// <param name="endDateTime">The ending datetime of the event</param>
        /// <param name="state">The state of the event</param>
        /// <param name="isVisible">Is the event visible</param>
        /// <param name="country">The country of the event</param>
        /// <param name="messages">The messages of the event</param>
        public Event(int id, string title, string description, string startDateTime, string endDateTime, string state,int isVisible,string country,Message[] messages)
        {
            Id = id;
            Title = title;
            Description = description;
            State = state;
            Country = country;
            StartDateTime = DateTime.Parse(startDateTime);
            if (endDateTime != null)
            {
                EndDateTime = DateTime.Parse(endDateTime);
            }
            Messages = new List<Message>();
        }

    }
}